# Comparing `tmp/fastq_handler-0.1.1.tar.gz` & `tmp/fastq_handler-0.1.3.tar.gz`

## filetype from file(1)

```diff
@@ -1 +1 @@
-gzip compressed data, was "fastq_handler-0.1.1.tar", max compression
+gzip compressed data, was "fastq_handler-0.1.3.tar", max compression
```

## Comparing `fastq_handler-0.1.1.tar` & `fastq_handler-0.1.3.tar`

### file list

```diff
@@ -1,10 +1,12 @@
--rw-r--r--   0        0        0    31904 2023-04-08 10:54:25.406907 fastq_handler-0.1.1/LICENSE
--rw-r--r--   0        0        0     1735 2023-04-09 21:29:09.259248 fastq_handler-0.1.1/README.md
--rw-r--r--   0        0        0      708 2023-04-09 21:38:13.624503 fastq_handler-0.1.1/pyproject.toml
--rw-r--r--   0        0        0      214 2023-04-08 11:18:53.731779 fastq_handler-0.1.1/src/fastq_handler/__init__.py
--rw-r--r--   0        0        0     1411 2023-04-09 21:28:47.254284 fastq_handler-0.1.1/src/fastq_handler/__main__.py
--rw-r--r--   0        0        0    14948 2023-04-06 16:59:47.945976 fastq_handler-0.1.1/src/fastq_handler/fastq_handler.py
--rw-r--r--   0        0        0     7660 2023-04-06 16:59:47.945976 fastq_handler-0.1.1/src/fastq_handler/records.py
--rw-r--r--   0        0        0     3984 2023-04-06 16:59:47.945976 fastq_handler-0.1.1/src/fastq_handler/utilities.py
--rw-r--r--   0        0        0     2752 2023-04-09 21:38:19.521533 fastq_handler-0.1.1/setup.py
--rw-r--r--   0        0        0     2428 2023-04-09 21:38:19.522091 fastq_handler-0.1.1/PKG-INFO
+-rw-r--r--   0        0        0    31904 2023-04-08 10:54:25.406907 fastq_handler-0.1.3/LICENSE
+-rw-r--r--   0        0        0     2393 2023-04-11 23:25:16.755777 fastq_handler-0.1.3/README.md
+-rw-r--r--   0        0        0      708 2023-04-11 23:23:04.423006 fastq_handler-0.1.3/pyproject.toml
+-rw-r--r--   0        0        0        0 2023-04-11 23:12:44.814007 fastq_handler-0.1.3/src/fastq_handler/__init__.py
+-rw-r--r--   0        0        0     1609 2023-04-11 23:22:46.582883 fastq_handler-0.1.3/src/fastq_handler/__main__.py
+-rw-r--r--   0        0        0     1488 2023-04-11 23:12:44.810006 fastq_handler-0.1.3/src/fastq_handler/actions.py
+-rw-r--r--   0        0        0     1068 2023-04-11 23:12:44.810006 fastq_handler-0.1.3/src/fastq_handler/configs.py
+-rw-r--r--   0        0        0    12595 2023-04-11 23:12:44.810006 fastq_handler-0.1.3/src/fastq_handler/fastq_handler.py
+-rw-r--r--   0        0        0     6758 2023-04-11 23:12:44.814007 fastq_handler-0.1.3/src/fastq_handler/records.py
+-rw-r--r--   0        0        0     3984 2023-04-11 23:12:44.814007 fastq_handler-0.1.3/src/fastq_handler/utilities.py
+-rw-r--r--   0        0        0     3414 2023-04-11 23:25:20.344028 fastq_handler-0.1.3/setup.py
+-rw-r--r--   0        0        0     3086 2023-04-11 23:25:20.344427 fastq_handler-0.1.3/PKG-INFO
```

### Comparing `fastq_handler-0.1.1/LICENSE` & `fastq_handler-0.1.3/LICENSE`

 * *Files identical despite different names*

### Comparing `fastq_handler-0.1.1/README.md` & `fastq_handler-0.1.3/README.md`

 * *Files 18% similar despite different names*

```diff
@@ -1,56 +1,60 @@
 # Fastq Handler
 
-A python module to process minion fastq files by concatenating reads as they are generated.
+[![PyPI version](https://badge.fury.io/py/fastq-handler.svg)](https://badge.fury.io/py/fastq-handler)
+[![PyPI pyversions](https://img.shields.io/pypi/pyversions/fastq-handler.svg)](https://pypi.python.org/pypi/fastq-handler/)
+[![PyPI license](https://img.shields.io/pypi/l/fastq-handler.svg)](https://pypi.python.org/pypi/fastq-handler/)
+[![PyPI status](https://img.shields.io/pypi/status/fastq-handler.svg)](https://pypi.python.org/pypi/fastq-handler/)
+[![PyPI format](https://img.shields.io/pypi/format/fastq-handler.svg)](https://pypi.python.org/pypi/fastq-handler/)
+
+A python module to process ONT fastq files by concatenating reads as they are generated during a sequencing run
 
 ## INTRODUCTION
 
-The _fastq_handler_ mmodule screens folders and subfolders for fastq files as they are generated by the MinION sequencer and concatenates them iteratively. This is useful for long reads that are split into multiple files. The output is a one file per
+The _fastqc-handler_ module screens folders and subfolders for fastq (fastq or fastq.gz format) files and concatenates them iteratively. This is useful for merging same-sample reads that are split into multiple files, as commonly obtained in ONT sequencing. The output is a one file per
 output fastq.gz, containing all reads from the previous files. The output directory structure is maintained.
 
 ## INPUT
 
-A directory containing fastq files. The files can be in subfolders. The files can be gzipped or not.
+A directory containing fastq files. The files can be in subfolders (each representing a different sample). The files can be gzipped or not.
 
 ## USAGE
 
 ```bash
-usage: fastq_handler.py [-h] [-i INPUT] [-o OUTPUT] [-n TAG] [--keep_names]
+usage: fastq_handler [-h] [-i INPUT] [-o OUTPUT] [-n TAG] [--keep_names]
 
 parse arguments
 
 optional arguments:
     -h, --help            show this help message and exit
     -i INPUT, --input INPUT
                         Input directory
     -o OUTPUT, --output OUTPUT
                         Output directory
     -n TAG, --tag TAG     Tag to add to output file name
     --keep_names          Keep original file names in output file
-    --monitor             run indefinitely
+    --max-size MAX_SIZE   max size of the output file, in kilobytes
 ```
 
 ## REQUIREMENTS
 
 **Modules**
 
 - dataclasses==0.6
 - natsort==8.3.1
 - pandas==1.5.3
-- pip==21.2.3
-- pytest==7.2.2
 - setuptools==57.4.0
 - xopen==1.7.0
 
 ## INSTALLATION
 
 ```bash
 python -m venv .venv
 source .venv/bin/activate
-python -m pip install fastq_handler
+python -m pip install fastq-handler
 ```
 
 ## MAIN OUTPUTS
 
 > **Note:** The output directory structure is maintained.
 
 - **fastq.gz** files containing all reads from the previous files.
```

### Comparing `fastq_handler-0.1.1/pyproject.toml` & `fastq_handler-0.1.3/pyproject.toml`

 * *Files 14% similar despite different names*

```diff
@@ -1,10 +1,10 @@
 [tool.poetry]
 name = "fastq_handler"
-version = "0.1.1"
+version = "0.1.3"
 description = "A python module to process minion fastq files by concatenating reads as they are generated"
 authors = ["SantosJGND <dourado.jns@gmail.com>", "Andre Santos <xiaodre2112@gmail.com>", "insapathogenomics <insapathogenomics@insa.min-saude.pt>"]
 license = "MIT"
 readme = "README.md"
 
 [tool.poetry.dependencies]
 python = "^3.9"
```

### Comparing `fastq_handler-0.1.1/src/fastq_handler/__main__.py` & `fastq_handler-0.1.3/src/fastq_handler/__main__.py`

 * *Files 14% similar despite different names*

```diff
@@ -1,12 +1,13 @@
 import argparse
 
 from fastq_handler.fastq_handler import PreMain
 
-from fastq_handler.records import ProcessActionMergeWithLast, RunConfig
+from fastq_handler.actions import ProcessActionMergeWithLast
+from fastq_handler.configs import RunConfig
 
 
 def get_arguments():
     parser = argparse.ArgumentParser(description="Process fastq files.")
     parser.add_argument(
         "-i", "--in_dir", help="Input directory", required=True)
     parser.add_argument("-o", "--out_dir",
@@ -20,28 +21,32 @@
 
     parser.add_argument(
         "--keep_names", help="keep original file names", action="store_true")
 
     parser.add_argument(
         "--monitor", help="run indefinitely", action="store_true")
 
+    parser.add_argument(
+        "--max-size", help="max size of the output file, in kilobytes", type=int, default=400000)
+
     return parser.parse_args()
 
 
 def main():
 
     args = get_arguments()
 
     run_metadata = RunConfig(
         fastq_dir=args.in_dir,
         output_dir=args.out_dir,
         name_tag=args.tag,
         actions=[ProcessActionMergeWithLast],
         keep_name=args.keep_names,
         sleep_time=args.sleep,
+        max_size=(args.max_size * 1000),
     )
 
     compressor = PreMain(
         run_metadata,
     )
 
     if args.monitor:
```

### Comparing `fastq_handler-0.1.1/src/fastq_handler/fastq_handler.py` & `fastq_handler-0.1.3/src/fastq_handler/fastq_handler.py`

 * *Files 20% similar despite different names*

```diff
@@ -8,15 +8,16 @@
 import os
 import subprocess
 import sys
 import time
 
 import pandas as pd
 
-from fastq_handler.records import Processed, RunConfig
+from fastq_handler.configs import RunConfig
+from fastq_handler.records import Processed
 from fastq_handler.utilities import Utils
 
 pd.options.mode.chained_assignment = None  # default='warn'
 
 
 """
 Notes:
@@ -245,14 +246,17 @@
 
         if not os.path.isfile(file_path):
             return False
 
         if self.match_to_processed(fastq_file, fastq_dir):
             return False
 
+        if self.processed.check_ignored(file_path):
+            return False
+
         return False if self.check_file_open(file_path) else True
 
     def get_files(self):
         """
         get folders and files
         """
 
@@ -363,27 +367,54 @@
 
     merged_dir_name = "merged_files"
     outfiles_dir_name = "out_files"
 
     def __init__(self, fastq_dir: str, run_metadata: RunConfig, processed: Processed, start_time: float):
         super().__init__(fastq_dir, run_metadata, processed, start_time)
 
+    def estimate_actions_size(self, fastq_file: str, sample_id: str) -> int:
+        """
+        estimate actions size
+        """
+
+        size = os.path.getsize(fastq_file)
+
+        if self.run_metadata.actions:
+
+            for process_action in self.run_metadata.actions:
+
+                size = process_action.estimate_output_size(
+                    size, sample_id, self.processed)
+
+        return size
+
     def process_file(self, fastq_file: str):
 
         destination_file = fastq_file
 
         if self.run_metadata.actions:
+
             destination_file = self.set_destination_filepath(
                 fastq_file, self.fastq_dir)
 
             self.append_to_file(fastq_file, destination_file)
 
             sample_id = self.processed.get_sample_id_from_merged(
                 destination_file)
 
+            projected_size = self.estimate_actions_size(
+                destination_file, sample_id)
+
+            if projected_size > self.run_metadata.max_size:
+                os.remove(destination_file)
+                self.processed.ignore_this(
+                    fastq_file
+                )
+                return self
+
             for process_action in self.run_metadata.actions:
 
                 process_action.process(
                     destination_file, sample_id, self.processed)
 
         self.update_processed(fastq_file, self.fastq_dir,
                               destination_file)
@@ -391,55 +422,17 @@
         return self
 
     def local_process(self):
 
         files_to_process = self.get_files()
 
         for ix, fastq_file in enumerate(files_to_process):
+
             self.process_file(fastq_file)
 
     def process_folder(self):
         """
         process folder, merge and update metadata
         submit to televir only the last file.
         """
         self.prep_output_dirs()
         self.local_process()
-
-############################ SYSTEM STUFF ##########################
-
-# TESTING
-
-# # print("barcoding on and gz")
-# minion_file_dir="C:\\Users\\andre\\OneDrive - FCT NOVA\\André\\Mestrado - Bioinfo\\2º Ano\\Projeto em Multi-Ómicas - INSA\\teste_1\\testing_automatic\\test_fastq_gz_bar\\fastq_gz_barcoding"
-# output_dir="q"
-# # output_dir="C:\\Users\\andre\\OneDrive - FCT NOVA\\André\\Mestrado - Bioinfo\\2º Ano\\Projeto em Multi-Ómicas - INSA\\teste_1\\testing_files\\testing_merging_and_metadata_files\\barcoded_samples"
-# tsv_temp_name="template_metadata.tsv"
-# tsv_temp_dir="C:\\Users\\andre\\OneDrive - FCT NOVA\\André\\Mestrado - Bioinfo\\2º Ano\\Projeto em Multi-Ómicas - INSA\\teste_1\\testing_files\\testing_merging_and_metadata_files\\barcoded_samples"
-# main(minion_file_dir, output_dir, tsv_temp_name, tsv_temp_dir, sleep_time=5)
-
-
-# # print("barcoding off and gz")
-# minion_file_dir="C:\\Users\\andre\\OneDrive - FCT NOVA\\André\\Mestrado - Bioinfo\\2º Ano\\Projeto em Multi-Ómicas - INSA\\teste_1\\testing_automatic\\test_fastq_gz_n_bar\\fastq_gz_n_barcoding"
-# output_dir="q"
-# # output_dir="C:\\Users\\andre\\OneDrive - FCT NOVA\\André\\Mestrado - Bioinfo\\2º Ano\\Projeto em Multi-Ómicas - INSA\\teste_1\\testing_files\\testing_merging_and_metadata_files\\barcoded_samples"
-# tsv_temp_name="template_metadata.tsv"
-# tsv_temp_dir="C:\\Users\\andre\\OneDrive - FCT NOVA\\André\\Mestrado - Bioinfo\\2º Ano\\Projeto em Multi-Ómicas - INSA\\teste_1\\testing_files\\testing_merging_and_metadata_files\\barcoded_samples"
-# main(minion_file_dir, output_dir, tsv_temp_name, tsv_temp_dir, sleep_time=5)
-
-
-# # print("barcoding on and fastq")
-# minion_file_dir="C:\\Users\\andre\\OneDrive - FCT NOVA\\André\\Mestrado - Bioinfo\\2º Ano\\Projeto em Multi-Ómicas - INSA\\teste_1\\testing_automatic\\test_fastq_bar\\fastq_barcoding"
-# output_dir="q"
-# # output_dir="C:\\Users\\andre\\OneDrive - FCT NOVA\\André\\Mestrado - Bioinfo\\2º Ano\\Projeto em Multi-Ómicas - INSA\\teste_1\\testing_files\\testing_merging_and_metadata_files\\barcoded_samples"
-# tsv_temp_name="template_metadata.tsv"
-# tsv_temp_dir="C:\\Users\\andre\\OneDrive - FCT NOVA\\André\\Mestrado - Bioinfo\\2º Ano\\Projeto em Multi-Ómicas - INSA\\teste_1\\testing_files\\testing_merging_and_metadata_files\\barcoded_samples"
-# main(minion_file_dir, output_dir, tsv_temp_name, tsv_temp_dir, sleep_time=5)
-
-
-# # print("barcoding off and fastq")
-# minion_file_dir="C:\\Users\\andre\\OneDrive - FCT NOVA\\André\\Mestrado - Bioinfo\\1º Ano\\2º Semestre\\Projeto em Multi-Ómicas - INSA\\teste_1\\testing_automatic\\test_fastq_n_bar\\fastq_n_barcoding"
-# output_dir="q"
-# # output_dir="C:\\Users\\andre\\OneDrive - FCT NOVA\\André\\Mestrado - Bioinfo\\2º Ano\\Projeto em Multi-Ómicas - INSA\\teste_1\\testing_files\\testing_merging_and_metadata_files\\barcoded_samples"
-# tsv_temp_name="template_metadata.tsv"
-# tsv_temp_dir="C:\\Users\\andre\\OneDrive - FCT NOVA\\André\\Mestrado - Bioinfo\\1º Ano\\2º Semestre\\Projeto em Multi-Ómicas - INSA\\teste_1\\testing_files\\testing_merging_and_metadata_files\\barcoded_samples"
-# main(minion_file_dir, output_dir, tsv_temp_name, tsv_temp_dir, sleep_time=5)
```

### Comparing `fastq_handler-0.1.1/src/fastq_handler/records.py` & `fastq_handler-0.1.3/src/fastq_handler/records.py`

 * *Files 21% similar despite different names*

```diff
@@ -1,18 +1,15 @@
 
 
 import os
-from abc import ABC, ABCMeta, abstractmethod
 from dataclasses import dataclass
 from typing import List, Optional, Type
 
 import pandas as pd
 
-from fastq_handler.utilities import Utils
-
 
 class Processed:
     """
     class to keep track of processed files
     """
     output_file = "processed.tsv"
 
@@ -27,14 +24,51 @@
         ]
     )
 
     def __init__(self, output_dir: str):
         self.output_dir = output_dir
         self.processed = self.read_processed()
 
+        self._ignore = []
+
+    @property
+    def processed_entries(self) -> pd.DataFrame:
+        """
+        get processed entries
+        """
+
+        processed_entries = self.processed[self.processed["merged"]
+                                           != self.processed["fastq"]].copy()
+
+        return processed_entries
+
+    def ignore_this(self, fastq_file: str):
+        """
+        ignore this file
+        """
+
+        self._ignore.append(fastq_file)
+
+    def check_ignored(self, fastq_file: str):
+        """
+        check if ignored
+        """
+
+        if fastq_file in self._ignore:
+            return True
+        else:
+            return False
+
+    def processed_fastq_list(self) -> List[str]:
+        """
+        get list of fastq files
+        """
+
+        return self.processed_entries.fastq.tolist()
+
     def read_processed(self):
         """
         read processed tsv
         """
 
         processed_file = os.path.join(
             self.output_dir,
@@ -226,88 +260,7 @@
                 "fastq",
                 "dir",
                 "barcode",
                 "time",
                 "merged",
             ]
         )
-
-
-class ProcessAction(ABCMeta):
-    """
-    abstract class for processing action
-    """
-
-    @staticmethod
-    @abstractmethod
-    def process(fastq_file: str, sample_id: str, processed: Processed):
-        """
-        process
-        """
-        pass
-
-
-class ProcessActionMergeWithLast(ProcessAction):
-    """get_dir_merged_last
-    class to merge with last"""
-
-    @staticmethod
-    def process(fastq_file: str, sample_id: str, processed: Processed):
-        """
-        process
-        """
-        last_run_file = processed.get_id_merged_last(
-            sample_id)
-
-        if last_run_file == "":
-            return
-
-        utils = Utils()
-
-        utils.append_file_to_gz(
-            last_run_file, fastq_file
-        )
-
-
-@dataclass
-class InputState:
-    """
-    class to hold input - defaults set for inheritance reasons. check py3.10 for better solution
-    """
-
-    fastq_dir: str = ""
-    name_tag: str = ""
-
-
-@dataclass
-class RunParams:
-    """
-    class to hold params with default values
-    """
-    actions: Optional[List[Type[ProcessAction]]] = None
-    keep_name: bool = False
-    sleep_time: int = 10
-
-
-@dataclass
-class OutputDirs:
-    """
-    class to hold output dirs
-    """
-    output_dir: str = "output"
-    logs_dirname: str = "logs"
-
-
-@dataclass
-class RunConfig(InputState, RunParams, OutputDirs):
-    """
-    class to hold run config"""
-
-    def __post_init__(self):
-        """
-        post init
-        """
-        if self.actions is None:
-            self.actions = []
-
-        self.logs_dir = os.path.join(self.output_dir, self.logs_dirname)
-        self.output_dir = os.path.abspath(self.output_dir)
```

### Comparing `fastq_handler-0.1.1/src/fastq_handler/utilities.py` & `fastq_handler-0.1.3/src/fastq_handler/utilities.py`

 * *Files identical despite different names*

### Comparing `fastq_handler-0.1.1/setup.py` & `fastq_handler-0.1.3/setup.py`

 * *Files 22% similar despite different names*

```diff
@@ -19,17 +19,17 @@
  'xopen==1.7.0']
 
 entry_points = \
 {'console_scripts': ['fastq_handler = fastq_handler.__main__:main']}
 
 setup_kwargs = {
     'name': 'fastq-handler',
-    'version': '0.1.1',
+    'version': '0.1.3',
     'description': 'A python module to process minion fastq files by concatenating reads as they are generated',
-    'long_description': '# Fastq Handler\n\nA python module to process minion fastq files by concatenating reads as they are generated.\n\n## INTRODUCTION\n\nThe _fastq_handler_ mmodule screens folders and subfolders for fastq files as they are generated by the MinION sequencer and concatenates them iteratively. This is useful for long reads that are split into multiple files. The output is a one file per\noutput fastq.gz, containing all reads from the previous files. The output directory structure is maintained.\n\n## INPUT\n\nA directory containing fastq files. The files can be in subfolders. The files can be gzipped or not.\n\n## USAGE\n\n```bash\nusage: fastq_handler.py [-h] [-i INPUT] [-o OUTPUT] [-n TAG] [--keep_names]\n\nparse arguments\n\noptional arguments:\n    -h, --help            show this help message and exit\n    -i INPUT, --input INPUT\n                        Input directory\n    -o OUTPUT, --output OUTPUT\n                        Output directory\n    -n TAG, --tag TAG     Tag to add to output file name\n    --keep_names          Keep original file names in output file\n    --monitor             run indefinitely\n```\n\n## REQUIREMENTS\n\n**Modules**\n\n- dataclasses==0.6\n- natsort==8.3.1\n- pandas==1.5.3\n- pip==21.2.3\n- pytest==7.2.2\n- setuptools==57.4.0\n- xopen==1.7.0\n\n## INSTALLATION\n\n```bash\npython -m venv .venv\nsource .venv/bin/activate\npython -m pip install fastq_handler\n```\n\n## MAIN OUTPUTS\n\n> **Note:** The output directory structure is maintained.\n\n- **fastq.gz** files containing all reads from the previous files.\n- **log.txt** file containing the concatenation process.\n\n## Maintainers\n\n- [**@xiaodre21**](https://github.com/xiaodre21)\n- [**@santosjgnd**](https://github.com/SantosJGND)\n- [**@insaflu**](https://github.com/insapathogenomics)\n',
+    'long_description': '# Fastq Handler\n\n[![PyPI version](https://badge.fury.io/py/fastq-handler.svg)](https://badge.fury.io/py/fastq-handler)\n[![PyPI pyversions](https://img.shields.io/pypi/pyversions/fastq-handler.svg)](https://pypi.python.org/pypi/fastq-handler/)\n[![PyPI license](https://img.shields.io/pypi/l/fastq-handler.svg)](https://pypi.python.org/pypi/fastq-handler/)\n[![PyPI status](https://img.shields.io/pypi/status/fastq-handler.svg)](https://pypi.python.org/pypi/fastq-handler/)\n[![PyPI format](https://img.shields.io/pypi/format/fastq-handler.svg)](https://pypi.python.org/pypi/fastq-handler/)\n\nA python module to process ONT fastq files by concatenating reads as they are generated during a sequencing run\n\n## INTRODUCTION\n\nThe _fastqc-handler_ module screens folders and subfolders for fastq (fastq or fastq.gz format) files and concatenates them iteratively. This is useful for merging same-sample reads that are split into multiple files, as commonly obtained in ONT sequencing. The output is a one file per\noutput fastq.gz, containing all reads from the previous files. The output directory structure is maintained.\n\n## INPUT\n\nA directory containing fastq files. The files can be in subfolders (each representing a different sample). The files can be gzipped or not.\n\n## USAGE\n\n```bash\nusage: fastq_handler [-h] [-i INPUT] [-o OUTPUT] [-n TAG] [--keep_names]\n\nparse arguments\n\noptional arguments:\n    -h, --help            show this help message and exit\n    -i INPUT, --input INPUT\n                        Input directory\n    -o OUTPUT, --output OUTPUT\n                        Output directory\n    -n TAG, --tag TAG     Tag to add to output file name\n    --keep_names          Keep original file names in output file\n    --max-size MAX_SIZE   max size of the output file, in kilobytes\n```\n\n## REQUIREMENTS\n\n**Modules**\n\n- dataclasses==0.6\n- natsort==8.3.1\n- pandas==1.5.3\n- setuptools==57.4.0\n- xopen==1.7.0\n\n## INSTALLATION\n\n```bash\npython -m venv .venv\nsource .venv/bin/activate\npython -m pip install fastq-handler\n```\n\n## MAIN OUTPUTS\n\n> **Note:** The output directory structure is maintained.\n\n- **fastq.gz** files containing all reads from the previous files.\n- **log.txt** file containing the concatenation process.\n\n## Maintainers\n\n- [**@xiaodre21**](https://github.com/xiaodre21)\n- [**@santosjgnd**](https://github.com/SantosJGND)\n- [**@insaflu**](https://github.com/insapathogenomics)\n',
     'author': 'SantosJGND',
     'author_email': 'dourado.jns@gmail.com',
     'maintainer': None,
     'maintainer_email': None,
     'url': None,
     'package_dir': package_dir,
     'packages': packages,
```

### Comparing `fastq_handler-0.1.1/PKG-INFO` & `fastq_handler-0.1.3/PKG-INFO`

 * *Files 16% similar despite different names*

```diff
@@ -1,10 +1,10 @@
 Metadata-Version: 2.1
 Name: fastq-handler
-Version: 0.1.1
+Version: 0.1.3
 Summary: A python module to process minion fastq files by concatenating reads as they are generated
 License: MIT
 Author: SantosJGND
 Author-email: dourado.jns@gmail.com
 Requires-Python: >=3.9,<4.0
 Classifier: License :: OSI Approved :: MIT License
 Classifier: Programming Language :: Python :: 3
@@ -16,61 +16,65 @@
 Requires-Dist: pip (==21.2.3)
 Requires-Dist: setuptools (==57.4.0)
 Requires-Dist: xopen (==1.7.0)
 Description-Content-Type: text/markdown
 
 # Fastq Handler
 
-A python module to process minion fastq files by concatenating reads as they are generated.
+[![PyPI version](https://badge.fury.io/py/fastq-handler.svg)](https://badge.fury.io/py/fastq-handler)
+[![PyPI pyversions](https://img.shields.io/pypi/pyversions/fastq-handler.svg)](https://pypi.python.org/pypi/fastq-handler/)
+[![PyPI license](https://img.shields.io/pypi/l/fastq-handler.svg)](https://pypi.python.org/pypi/fastq-handler/)
+[![PyPI status](https://img.shields.io/pypi/status/fastq-handler.svg)](https://pypi.python.org/pypi/fastq-handler/)
+[![PyPI format](https://img.shields.io/pypi/format/fastq-handler.svg)](https://pypi.python.org/pypi/fastq-handler/)
+
+A python module to process ONT fastq files by concatenating reads as they are generated during a sequencing run
 
 ## INTRODUCTION
 
-The _fastq_handler_ mmodule screens folders and subfolders for fastq files as they are generated by the MinION sequencer and concatenates them iteratively. This is useful for long reads that are split into multiple files. The output is a one file per
+The _fastqc-handler_ module screens folders and subfolders for fastq (fastq or fastq.gz format) files and concatenates them iteratively. This is useful for merging same-sample reads that are split into multiple files, as commonly obtained in ONT sequencing. The output is a one file per
 output fastq.gz, containing all reads from the previous files. The output directory structure is maintained.
 
 ## INPUT
 
-A directory containing fastq files. The files can be in subfolders. The files can be gzipped or not.
+A directory containing fastq files. The files can be in subfolders (each representing a different sample). The files can be gzipped or not.
 
 ## USAGE
 
 ```bash
-usage: fastq_handler.py [-h] [-i INPUT] [-o OUTPUT] [-n TAG] [--keep_names]
+usage: fastq_handler [-h] [-i INPUT] [-o OUTPUT] [-n TAG] [--keep_names]
 
 parse arguments
 
 optional arguments:
     -h, --help            show this help message and exit
     -i INPUT, --input INPUT
                         Input directory
     -o OUTPUT, --output OUTPUT
                         Output directory
     -n TAG, --tag TAG     Tag to add to output file name
     --keep_names          Keep original file names in output file
-    --monitor             run indefinitely
+    --max-size MAX_SIZE   max size of the output file, in kilobytes
 ```
 
 ## REQUIREMENTS
 
 **Modules**
 
 - dataclasses==0.6
 - natsort==8.3.1
 - pandas==1.5.3
-- pip==21.2.3
-- pytest==7.2.2
 - setuptools==57.4.0
 - xopen==1.7.0
 
 ## INSTALLATION
 
 ```bash
 python -m venv .venv
 source .venv/bin/activate
-python -m pip install fastq_handler
+python -m pip install fastq-handler
 ```
 
 ## MAIN OUTPUTS
 
 > **Note:** The output directory structure is maintained.
 
 - **fastq.gz** files containing all reads from the previous files.
```

